### ISSUE TITLE
graphroot and rootless_storage_path are ignored in global storage.conf

### ISSUE THREAD

**Issue by plopresti** [containers/podman#7876]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

Somewhat similar to #7513... I am unable to set the graphroot (or rootless_storage_path) in the global storage.conf file.

**Steps to reproduce the issue:**

1. rm ~/.config/containers/storage.conf

2. Edit /etc/containers/storage.conf to set graphroot = "/scratch/$USER/containers/storage" (or anything really)

3. Run "podman info | grep scratch"

**Describe the results you received:**

Empty output.

**Describe the results you expected:**

  graphRoot: /scratch/ploprest/containers/storage
  volumePath: /scratch/ploprest/containers/storage/volumes

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `podman version`:**

```
Version:      2.1.1
API Version:  2.0.0
Go Version:   go1.15
Built:        Sun Sep 27 19:07:17 2020
OS/Arch:      linux/amd64
```

**Output of `podman info --debug`:**

```
host:
  arch: amd64
  buildahVersion: 1.16.1
  cgroupManager: cgroupfs
  cgroupVersion: v1
  conmon:
    package: conmon-2.0.21-1.el7.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.0.21, commit: efe288263ebb1bab4673d2cfbe1a352f3c7d27c1-dirty'
  cpus: 64
  distribution:
    distribution: '"scientific"'
    version: "7.8"
  eventLogger: journald
  hostname: vex4
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 100
      size: 1
    uidmap:
    - container_id: 0
      host_id: 39327
      size: 1
  kernel: 3.10.0-1127.19.1.el7.x86_64
  linkmode: dynamic
  memFree: 302913359872
  memTotal: 404159303680
  ociRuntime:
    name: runc
    package: runc-1.0.0-103.dev.el7.x86_64
    path: /usr/bin/runc
    version: |-
      runc version 1.0.0-rc10
      commit: 70f396b2cfa8fb0c6c12bd46020af4744b9eff5b
      spec: 1.0.1-dev
  os: linux
  remoteSocket:
    path: /run/user/39327/podman/podman.sock
  rootless: true
  slirp4netns:
    executable: /bin/slirp4netns
    package: slirp4netns-1.1.4-2.el7.x86_64
    version: |-
      slirp4netns version 1.1.4
      commit: b66ffa8e262507e37fca689822d23430f3357fe8
      libslirp: 4.3.1
      SLIRP_CONFIG_VERSION_MAX: 3
  swapFree: 17162039296
  swapTotal: 17179865088
  uptime: 864h 15m 42.93s (Approximately 36.00 days)
registries:
  search:
  - registry.fedoraproject.org
  - registry.access.redhat.com
  - registry.centos.org
  - docker.io
store:
  configFile: /home/ploprest/.config/containers/storage.conf
  containerStore:
    number: 0
    paused: 0
    running: 0
    stopped: 0
  graphDriverName: overlay
  graphOptions:
    overlay.ignore_chown_errors: "true"
    overlay.mount_program:
      Executable: /bin/fuse-overlayfs
      Package: fuse-overlayfs-1.1.2-1.el7.x86_64
      Version: |-
        fusermount3 version: 3.6.1
        fuse-overlayfs: version 1.1.0
        FUSE library version 3.6.1
        using FUSE kernel interface version 7.29
  graphRoot: /home/ploprest/.local/share/containers/storage
  graphStatus:
    Backing Filesystem: nfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
    Using metacopy: "false"
  imageStore:
    number: 0
  runRoot: /run/user/39327/containers
  volumePath: /home/ploprest/.local/share/containers/storage/volumes
version:
  APIVersion: 2.0.0
  Built: 1601258837
  BuiltTime: Sun Sep 27 19:07:17 2020
  GitCommit: ""
  GoVersion: go1.15
  OsArch: linux/amd64
  Version: 2.1.1

```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

```
(paste your output here)
```

**Have you tested with the latest version of Podman and have you checked the Podman Troubleshooting Guide?**

Yes

**Additional environment details (AWS, VirtualBox, physical, etc.):**

Physical system running CentOS 7.8

**Comment by rhatdan** [comment]
Can you try this on a brand new user account?  Or you can do 
```
$ podman system reset
$ podman system migrate 
$ podman info 
```
Does it change?

**Comment by plopresti** [comment]
@rhatdan Same result. Are you unable to reproduce?

**Comment by rhatdan** [comment]
Have not tried yet.  I attempted to fix this earlier, but maybe I missed something, (Most likely)

**Comment by rcasta74** [comment]
@rhatdan I confirm this issue: bug #7546 is not fixed.
Updated today to version 2.1.1, but if I remove `~/.config/containers/storage.conf` I get:
```
$ podman info | fgrep graphRoot
  graphRoot: /home/rcastagnola/.local/share/containers/storage
```
while `/etc/containers/storage.conf` contains:
```
$ grep ^[^#] /etc/containers/storage.conf
[storage]
driver = "overlay"
runroot = "/var/run/containers/storage"
graphroot = "/mnt/containers/storage"
rootless_storage_path = "/mnt/containers/storage-$UID"
[storage.options]
additionalimagestores = [
]
[storage.options.overlay]
mountopt = "nodev,metacopy=on"
[storage.options.thinpool]

```

**Comment by plopresti** [comment]
@rhatdan Thanks.

Incidentally, #7879 is more of a showstopper for me... This bug has a work-around; i.e. I can ask each user to edit .config/containers/storage.conf. #7879 has no work-around at all, and it renders additionalimagestores unusable.

**Comment by rcasta74** [comment]
My understanding from  looking at the code is that the `/etc/containers/storage.conf` file is never read when run in rootless mode, so no way to get it to work.
What I would like to understand is what is the wanted behaviour: should `/etc/containers/storage.conf` file be used to configure the storage for rootless mode as well, or should it be kept separated?
In the latter case it doesn't make sense to me to have a different config name to override the default value: `graphroot` could be used instead of `rootless_storage_path`

**Comment by rcasta74** [comment]
Sorry for my last comment, I was wrong: I totally missed that the file is parsed into init function.

Anyway, adding few Printf commands I found out where the issue is.
Into the function `defaultStoreOptionsIsolated` (utils.go), the variable err before the last if block (`if rootless && rootlessUID != 0`) contains:
`err: stat /home/roberto/.config/containers/storage.conf: no such file or directory`
because the user conf is missing, thus the block is bypassed and the RootlessStoragePath value is not used to change GraphRoot value.

I have tryed then to create the file `~/.config/containers/storage.conf` without redefining the property `rootless_storage_path`, but in this case the variable `storageOpts.RootlessStoragePath` became empty because of the code:
```
storageOpts = StoreOptions{}
reloadConfigurationFileIfNeeded(storageConf, &storageOpts)
```
thus creating the user conf file the RootlessStoragePath value is forgotten.
